Vizzuality/mangrove-atlas
# Gmw 460 fe implement ability to see the suggestions for contextual layers or satellite basemaps

##  Map Settings added to the sidebar

### Overview

This feature has different steps:
- Removes basemap selector from the map legend 
- Add MapSettings button to sidebar
- Adds basemaps removed from the legend to the Map Settings section from the sidebar
- Creates a proxy to connect with Planet API
- Adds a new env bar NEXT_PUBLIC_PLANET_API_KEY
- Some widgets will have suggested basemaps for better visualization of the layer. Currently is just included in the alerts widget

### Designs

_[Link to the related design prototypes (if applicable)](https://www.figma.com/file/GJCx8yGfx6JERPuxlpG3nK/Mangrove-Watch-%5BInternal%5D?node-id=2559%3A18838&mode=dev)_

### Testing instructions

Change basemaps in mapsettings
Open alerts widget and change basemap

Expected behavior of this tool:
- User could select just one of the previous basemaps (light, dark, and satellite) at a time -> mapStyle for Map. 
- User should be able to select one contextual satellite basemaps at a time. The user should have visual feedback on which one is selected
- When a contextual map is selected, the component should show extra features for the basemap, in this case, a dropdown with available dates
- Map settings should be disable when drawing tool is open. The reason is to prevent the user to go from custom area to map settings to avoid  not having a location_id available


### Feature relevant tickets

_[Link to the related task manager tickets](https://vizzuality.atlassian.net/browse/GMW-459?atlOrigin=eyJpIjoiY2UxM2RhNzNkZmNjNGI1OGFkMzI3YmE2NWQ1N2FmZDQiLCJwIjoiaiJ9)_

---

## Checklist before submitting

      deploying to staging/production, please add brief testing instructions
      to the deploy checklist
